### ISSUE TITLE
Binding Action for other object

### ISSUE THREAD

**Issue by enzyme69** [samytichadou/puppeteer_blender_addon#1]
Hi Samy, love this addon, I have been waiting for something like this, kind of like a remote control key binding, it might be useful also to do like DJ timing to music, so I wonder if the addon can have feature like:
- ability to change the parent puppet for the same action animation. if I start with Suzanne, but I have other objects, like buttons to do the same motion, maybe if the action can be duplicated and reassigned into different parent?
- ability to bind sound into the action?
- automation set can be run at the same time (I have monkey A, monkey B, monkey C, I thought that the set is for each object, but can be active all at the same time)

**Comment by samytichadou** [comment]
Hi, thanks for the kind words and cool to hear you like the addon ! :)

- ability to change the parent puppet for the same action animation : this is kind of tricky, the addon does not use selected and active object(s) to get which object has to be animated, instead it looks at the name (and the name of the action if `paste mode` setup to "action" or "parent and action"). How could the user tell blender which object to look at in your suggestion ? selection ? but an automation could be using 10 different objects  (and only 1 automation entry) so all animations would not be attached to only 1 object... Sorry if i'm not clear but do you get what i mean ?
- bind sound to action : that would be possible with a trigger system, automation would be triggered by some changes or datas (sound amplitude...) but i'm completly noob in sound handling in python, and giving it is not the initial purpose of the addon, i don't really have the time to add this big part right now, definitely something to keep in mind though ! (would love to have this ability for some of my personal projects too !)
- automation set can be run at the same time : What would be the point of this ? Maybe it was not really clear in the how to video, but you don't have to have different set for each object you animate. Just create a "monkeys" set, and put all the automations in it. A set can contains as much automations as you want, and an automation can contains as much keyframes (for as much objects or datas) as you want, so you can just put all the automations in the same set, am i right ? The whole set idea is to organize the file if you have a lot of automations going on, but it creates some "separate runtime" to help work without being overwhelmed by the number of bindings to handle. Does that make sense ? I could add some "all set" mode, but i don't really see the use case...

Thanks a lot for suggestions and interest in the addon !

**Comment by enzyme69** [comment]
Cheers for detailed reply. 

Ok I need to simplify the request :) Basically I might be thinking that your automation addon can be used like for "realtime" control that trigger animation, for simple case like DJ-ing. I have to test this using a few "action" on a single object and multiple objects first.

As for sound, I think in the past I have been experimenting with MIDI to control basic motion. But with something like your addon, potentially we can trigger "action" based on MIDI or OSC. But MIDI is probably simpler. 

I will study this addon first. I think changing and assigning action into custom X object would be the highly wanted feature. What I mean is that:
- I started with Suzanne for example, I made action automation A, B, C, etc.
- Then suddenly I want the same action for Suzanne B. How to transfer this animation?

**Comment by samytichadou** [comment]
Yeah i see what you mean about dj ing application. And would be a cool use for puppeteer, just need to find some time to wrap my head around MIDI integration (never done something like that !)

And about applying action to other object, you kind of can, set the `paste mode` to "Action" and keyframes will just be pasted in original actions. Like any other blender datablock, share the action between several objects.

You can also edit directly the puppeteer keyframes data in the outliner, datablock mode. Interesting prop would be parent_name and action_name, but puppeteer also use object type (and some kind of subtype) to locate soon-to-be-animated fcurves, so this may break regarding your animation (posemode anim transferred to a mesh object without armature for example)

For now puppeteer is meant as something recording "absolute" action, i thought the nature of action (and its target object(s)) should be let to the user to deal with. But it may be something interesting. 
How would you deal with it practically giving the current state of the addon ?

**Comment by enzyme69** [comment]
I noticed the "Additive Mode" but have not tried the paste option. They act as keyframe, maybe some people also prefer the NLA way kind of blending and layering action, but can be triggered via button. 

I will make video soon about this, I think this can be super useful to make like a drummer type of animation but hands on :) It's fun, I just test the timing.

Someone mentioned that maybe using Game Controller like PS5 support will also be awesome.

To make it more practical and fast interaction is like I said if we can "transfer" the parent or easily make duplicate of same Set of Animation Triggers into different object. --> hope I explain correctly.
